src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,214 @@
+                 PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.setLong(3, car.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get car from database", e);
+        }
+        insertDrivers(car);
+        deleteCarsDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = FALSE "
+                + "WHERE id = (?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get car from database", e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> list = new ArrayList<>();
+        String query = "SELECT model, manufacturer_id FROM cars JOIN cars_drivers "
+                + "ON cars.id = cars_drivers.car_id WHERE cars_drivers.driver_id = (?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                list.add(getCar(resultSet));
+            }
+            for (Car car : list) {
+                car.setDrivers(getDriversOfCar(car.getId()));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get car from database with driver id: "
+                    + driverId, e);
+        }
+        return list;
+    }
+
+    private List<Driver> getDriversOfCar(Long carId) {
+        List<Driver> list = new ArrayList<>();
+        String query = "SELECT * FROM drivers JOIN cars_drivers "
+                + "ON drivers.id = cars_drivers.driver_id WHERE cars_drivers.car_id = (?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, carId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                list.add(getDrivers(resultSet));
+            }
+            return list;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get drivers from database with car id: "
+                    + carId, e);
+        }
```

Comment:
i think will be better if you write `List<Driver> list = new ArrayList<>();` before the `while (resultSet.next()) {`. also list is not so informative name, drivers or smth like that is better

Reply:
I don`t think that this is a mistake